gradient/cli/common.py

```diff
@@ -171,3 +219,31 @@ def options_file(f):
         )
     ]
     return functools.reduce(lambda x, opt: opt(x), reversed(options), f)
+
+class GradientObjectListOption(GradientOption):
+    def __init__(self, object_name, param_decls, **kwargs):
+        super(GradientObjectListOption, self).__init__(param_decls, **kwargs)
+        self.object_name = object_name
+        self.object_list_name = self.object_name + "s"
+        self.object_key = self._compose_object_key()
+
+    # Get name of overall object list for this type of option
+    def get_object_list_name(self):
+        return self.object_list_name
+
+    # Get this option's object key for the objects in the object list
+    def _compose_object_key(self):
+        option_name = get_option_name(self.opts)
+        if not option_name.startswith(self.object_name):
+            return option_name
+
+        object_key = option_name[len(self.object_name):]
+        object_key = object_key[0].lower() + object_key[1:]
+        return object_key
+
+    def get_object_key(self):
```

Comment:
Similar to the one above - can be removed. Just directly read the instance's field